plugin/trino-iceberg/pom.xml

```diff
@@ -406,6 +406,7 @@
                         <ignoredResourcePattern>about.html</ignoredResourcePattern>
                         <ignoredResourcePattern>iceberg-build.properties</ignoredResourcePattern>
                         <ignoredResourcePattern>mozilla/public-suffix-list.txt</ignoredResourcePattern>
+                        <ignoredResourcePattern>google/protobuf/.*proto</ignoredResourcePattern>
```

Comment:
Why is it safe to load random `google/protobuf/.*proto` files?
is Trino behavior going to be non-deterministic?

Reply:
the same is already done in other places in the project, so it seemed reasonable to do the same here as well: https://github.com/trinodb/trino/blob/55f306bfc6d126a76c949cc29c2c7c2cca673ed7/plugin/trino-kafka/pom.xml#L338 / https://github.com/trinodb/trino/blob/450c9ad4d1ee0dfe5a0c06b02128b3c67ce87646/plugin/trino-kinesis/pom.xml#L195 / https://github.com/trinodb/trino/blob/450c9ad4d1ee0dfe5a0c06b02128b3c67ce87646/plugin/trino-redis/pom.xml#L213

Reply:
@nastra i don't know how/whether Redis, Kafka, Kinesis and Redis connectors use Protocol Buffers, so I don't know whether I should be worried about them. I think ORC uses Protocol Buffers for it's file footer format, and Iceberg uses ORC, so I am worried about Iceberg, until someone explains me why i should not be.

Reply:
what would be your suggested approach to this?

Reply:
Can we strip them out when building the `orc-core-nohive` module?

Reply:
what are the `google/protobuf/.*\.proto` resources in that jar?

Reply:
these are the ones that are shipped within the orc-nohive.jar that are conflicting
```
[WARNING]   google/protobuf/any.proto
[WARNING]   google/protobuf/api.proto
[WARNING]   google/protobuf/compiler/plugin.proto
[WARNING]   google/protobuf/descriptor.proto
[WARNING]   google/protobuf/duration.proto
[WARNING]   google/protobuf/empty.proto
[WARNING]   google/protobuf/field_mask.proto
[WARNING]   google/protobuf/source_context.proto
[WARNING]   google/protobuf/timestamp.proto
[WARNING]   google/protobuf/type.proto
```

Reply:
@electrum what do you think?

Reply:
@findepi @electrum Any possibility to move this PR forward? I don't think we should have the `.proto` files in there in the first place (and they will be removed in ORC 1.8.2). The proto files are only used at compile time, which happens in the ORC release process, so they are redundant and we can safely exclude them.

Reply:
So i now understand at least what these proto files are for. These are for builtin Protobuf types such as https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#timestamp.
I do hope nothing depends on them, but it still is something to address.



> and they will be removed in ORC 1.8.2

do you have a link to an ORC issue/pr addressing this?

let's have a Trino issue, to remove these will-be-redundant exclusions
and let's have a TODO code comment linking to the Trino issue.
The Trino issue should link to the ORC issue/pr.

Reply:
@findepi It is improbable that something depends on them because they are only used when compiling the resources. This is not something that you typically do in a downstream project. This is also why they'll be removed in 1.8.2. 

I created a ticket: https://github.com/trinodb/trino/issues/15371 And referenced the issue where we exclude the proto files, so we can clean this up once 1.8.2 has been released.